app/src/main/java/com/github/epfl/meili/posts/PostActivity.kt

```diff
@@ -90,15 +117,42 @@ class PostActivity : AppCompatActivity() {
 
         viewModel.initDatabase(FirestoreDatabase("forum/$postId/comments", Comment::class.java))
         viewModel.getElements().observe(this, { map ->
-            recyclerAdapter.submitList(map.toList())
-            recyclerAdapter.notifyDataSetChanged()
+            commentListener(map)
         })
     }
 
+    private fun commentListener(commentsMap: Map<String, Comment>) {
+        val newUsers = ArrayList<String>()
+        for ((commentId, comment) in commentsMap) {
+            newUsers.add(comment.authorUid)
+        }
+
+        serviceProvider().getUserInformation(newUsers, { onUsersInfoReceived(it, commentsMap) },
+            { Log.d(TAG, "Error when fetching users information") })
+    }
+
+    override fun onUsersInfoReceived(users: Map<String, User>, commentsMap: Map<String, Comment>) {
+        Log.d(TAG, "on users info received")
+        Log.d(TAG, commentsMap.toString())
+        Log.d(TAG, users.toString())
+        usersMap = HashMap(usersMap) + users
+        val commentsAndUsersMap = HashMap<String, Pair<Comment, User>>()
+        for ((commentId, comment) in commentsMap) {
+            val user = usersMap[comment.authorUid]
+            Log.d(TAG, user.toString())
+            if (user != null) {
+                commentsAndUsersMap[commentId] = Pair(comment, user)
+            }
+        }
+
+        recyclerAdapter.submitList(commentsAndUsersMap.toList())
+        recyclerAdapter.notifyDataSetChanged()
+    }
+
     private fun initRecyclerAdapter() {
-        recyclerAdapter = CommentsRecyclerAdapter(viewModel)
+        recyclerAdapter = CommentsRecyclerAdapter(this)
         val recyclerView: RecyclerView = findViewById(R.id.comments_recycler_view)
-        initRecyclerView(recyclerAdapter, recyclerView,this)
+        initRecyclerView(recyclerAdapter, recyclerView, this)
         ViewCompat.setNestedScrollingEnabled(recyclerView, false);
```

Comment:
Nitpick: redundant semicolon

Reply:
This line wasn't added in this PR but I'll change that. Thanks!